### PR TITLE
Patch update analysis

### PR DESCRIPTION
**Pull request by seallard**
## Description
Patches a broken endpoint in the web api for updating the analysis from cigrid. Closes https://github.com/Clinical-Genomics/trailblazer/issues/303.

### Fixed
- Patch broken endpoint for updating an analysis

### How to test
- [ ] deploy to web stage
- [ ] verify that the status/visibility and comment can be updated.

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: trailblazer/store/crud/update.py [hunk 1]
```diff
@@ -221,3 +221,21 @@ def update_analysis_comment(self, case_id: str, comment: str) -> None:
...
+
+        if "comment" in data:
+            comment = str(data["comment"])
+            LOG.info(f"Adding comment {comment} to analysis {analysis.family}")
+            analysis.comment = comment
+
+        if "is_visible" in data:
+            is_visibile = bool(data["is_visible"])
+            LOG.info(f"Setting visibility to {is_visibile} for analysis {analysis.family}")
+            analysis.is_visible = is_visibile
```
Comment (islean) [inline trailblazer/store/crud/update.py:237]:
```suggestion
            analysis.is_visible = is_visible
```

File: trailblazer/server/api.py [hunk 2]
```diff
@@ -72,13 +72,13 @@ def analyses():
...
+    """Retrieve or update an analysis."""
     analysis: Analysis = store.get_analysis_with_id(analysis_id)
     if analysis is None:
         return abort(404)
 
     if request.method == "PUT":
-        analysis.update(request.json)
+        store.update_analysis(analysis_id=analysis_id, data=request.json)
 
     data = analysis.to_dict()
```
Comment (islean) [inline trailblazer/server/api.py:83]: slightly unfortunate with the naming clash here. Might be worth to change the former to "body" or something but feel free to disregard.

File: trailblazer/server/api.py [hunk 3]
```diff
@@ -72,13 +72,13 @@ def analyses():
...
 @blueprint.route("/analyses/<int:analysis_id>", methods=["GET", "PUT"])
 def analysis(analysis_id):
-    """Display a single analysis."""
+    """Retrieve or update an analysis."""
     analysis: Analysis = store.get_analysis_with_id(analysis_id)
     if analysis is None:
         return abort(404)
 
     if request.method == "PUT":
-        analysis.update(request.json)
```
Comment (islean) [inline trailblazer/server/api.py:81]: Was there never an Analysis.update method at all?

**Review by islean — APPROVED** [review]
Looks good!

File: trailblazer/server/api.py (hunk 3 above)
Comment (islean) [reply trailblazer/server/api.py:81]: Nevermind, saw in the issue.

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

File: trailblazer/store/crud/update.py [hunk 4]
```diff
@@ -221,3 +221,21 @@ def update_analysis_comment(self, case_id: str, comment: str) -> None:
             " ".join([analysis.comment, comment]) if analysis.comment else comment
         )
         LOG.info(f"Adding comment {comment} to analysis {analysis.family}")
+
+    def update_analysis(self, analysis_id: int, data: dict) -> Analysis:
```
Comment (ChrOertlin) [inline trailblazer/store/crud/update.py:225]: we should add a test for this

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by seallard** [comment]
Tested in stage, fields can be edited as expected

**Pull request merged**